src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,189 @@
+package duke;
+
+
+/**
+ * Represents the user interface for the Duke program, providing methods for
+ * displaying messages and information to the user.
+ */
+public class Ui {
+
+    /**
+     * Displays the Duke logo and an initial message to the user.
+     */
+    public void message() {
+        // Displaying Duke logo and initial message
+        String logo = " KASSIM ";
+        System.out.println("YOO I AM " + logo);
+        System.out.println("What can I do for you?");
+    }
+
+    /**
+     * Displays an error message encountered during the Duke program execution.
+     *
+     * @param e The DukeException containing the error message.
+     */
+    public void errorEncounter(DukeException e) {
+        System.out.println("------------------------------------------------------------");
+        System.out.println(e.getMessage());
+        System.out.println("------------------------------------------------------------");
+    }
+
+    /**
+     * Returns a message indicating that a task description is empty.
+     *
+     * @return The error message for an empty task description.
+     */
+    public String emptyErrorMessage() {
+        return "yes u have your task, but what your task plan to do? please indicate it.";
+    }
+
+    /**
+     * Returns a general error message for unexpected inputs.
+     *
+     * @return The general error message for unexpected inputs.
+     */
+    public String errorMessage() {
+        return "what? please check your input.";
+    }
+
+    /**
+     * i an error message for an unrecognized command.
+     */
+    public void commandError() {
+        System.out.println("SORRY! but are you sure you enter the correct command? please check!");
+    }
+
+    /**
+     * Displays information that a task has been marked as done
+     *
+     * @param task The task that has been marked as done.
+     */
+    public void markInfo(Task task) {
+        System.out.println("------------------------------------------------------------");
```

Comment:
Consider using a constant for the separation line to ensure consistency across different methods.

Example:
private static final String SEPARATION_LINE = "------------------------------------------------------------";

Doing this, you won't have to copy/paste the line over and over again and can just
System.out.println(SEPARATION_LINE);